internal/provider/credential_provider_resource.go

```diff
@@ -225,6 +231,81 @@ func (r *credentialProviderResource) Schema(_ context.Context, _ resource.Schema
 					},
 				},
 			},
+			"oauth_authorization_code": schema.SingleNestedAttribute{
+				Description: "OAuth Authorization Code Flow type Credential Provider configuration.",
+				Optional:    true,
+				Attributes: map[string]schema.Attribute{
+					"oauth_url": schema.StringAttribute{
+						Description: "OAuth URL for the OAuth Credential Provider.",
+						Required:    true,
+					},
+					"authorization_url": schema.StringAttribute{
+						Description: "Authorization URL for the OAuth Credential Provider.",
+						Required:    true,
+					},
+					"token_url": schema.StringAttribute{
+						Description: "Token URL for the OAuth Credential Provider.",
+						Required:    true,
+					},
+					"client_id": schema.StringAttribute{
+						Description: "Client ID for the OAuth Credential Provider.",
+						Required:    true,
+					},
+					"client_secret": schema.StringAttribute{
+						Description: "Client Secret for the OAuth Credential Provider.",
+						Required:    true,
+						Sensitive:   true,
+					},
+					"scopes": schema.StringAttribute{
+						Description: "Scopes for the OAuth Credential Provider.",
+						Optional:    true,
+					},
+					"custom_parameters": schema.SetNestedAttribute{
+						Description: "Set Custom Parameters for the OAuth Credential Provider.",
+						Optional:    true,
+						NestedObject: schema.NestedAttributeObject{
+							Attributes: map[string]schema.Attribute{
+								"key": schema.StringAttribute{
+									Description: "Key for Custom Parameter of the OAuth Credential Provider.",
+									Required:    true,
+								},
+								"value": schema.StringAttribute{
+									Description: "Value for Custom Parameter of the OAuth Credential Provider.",
+									Required:    true,
+								},
+								"value_type": schema.StringAttribute{
+									Description: "Type of value for Custom Parameter of the OAuth Credential Provider. Possible values are `literal` or `dynamic`.",
+									Optional:    true,
+									Computed:    true,
+									Default:     stringdefault.StaticString("literal"),
+									Validators: []validator.String{
+										stringvalidator.OneOf([]string{
+											"literal",
+											"dynamic",
+										}...),
+									},
+								},
+							},
+						},
+					},
+					"is_pkce_required": schema.BoolAttribute{
+						Description: "PKCE required flag for the OAuth Credential Provider.",
+						Optional:    true,
+					},
+					"callback_url": schema.StringAttribute{
+						Description: `Authorization Callback URL for the OAuth Credential Provider. Callback URL can be pre-generated by matching the following template; 
```

Comment:
Please format the description similarly to how I did in the Client workload so that it maps better on the Markdown file
https://github.com/Aembit/terraform-provider-aembit/blob/main/internal/provider/client_workload_resource.go#L74-L93